vmm/src/vm.rs

```diff
@@ -716,12 +725,72 @@ impl Vm {
             vm.sev_snp_init().map_err(Error::InitializeSevSnpVm)?;
         }
 
-        #[cfg(feature = "fw_cfg")]
-        device_manager
-            .lock()
-            .unwrap()
-            .create_fw_cfg_device()
-            .map_err(Error::DeviceManager)?;
+        #[cfg(all(feature = "fw_cfg", not(target_arch = "riscv64")))]
+        {
+            device_manager
```

Comment:
Why not grab the lock of `device_manager` once? I know this pattern is used multiple times in CHV, but I don't think it is optimal